crates/cli/src/model/datadog_api.rs

```diff
@@ -242,12 +247,93 @@ pub struct StaticAnalysisRulesAPIResponse {
     pub data: APIResponseRuleset,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct SecretRuleApiMatchValidationHttpCode {
+    pub start: u16,
+    pub end: u16,
+}
+
+impl From<SecretRuleApiMatchValidationHttpCode> for SecretRuleMatchValidationHttpCode {
+    fn from(value: SecretRuleApiMatchValidationHttpCode) -> Self {
+        SecretRuleMatchValidationHttpCode {
+            start: value.start,
+            end: value.end,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum SecretRuleApiMatchValidationHttpMethod {
+    #[serde(rename = "GET")]
+    Get,
+    #[serde(rename = "POST")]
+    Post,
+    #[serde(rename = "PATCH")]
+    Patch,
+    #[serde(rename = "PUT")]
+    Put,
+    #[serde(rename = "DELETE")]
+    Delete,
+}
+
+impl From<SecretRuleApiMatchValidationHttpMethod> for SecretRuleMatchValidationHttpMethod {
+    fn from(val: SecretRuleApiMatchValidationHttpMethod) -> Self {
+        match val {
+            SecretRuleApiMatchValidationHttpMethod::Get => SecretRuleMatchValidationHttpMethod::Get,
+            SecretRuleApiMatchValidationHttpMethod::Post => {
+                SecretRuleMatchValidationHttpMethod::Post
+            }
+            SecretRuleApiMatchValidationHttpMethod::Put => SecretRuleMatchValidationHttpMethod::Put,
+            SecretRuleApiMatchValidationHttpMethod::Patch => {
+                SecretRuleMatchValidationHttpMethod::Patch
+            }
+            SecretRuleApiMatchValidationHttpMethod::Delete => {
+                SecretRuleMatchValidationHttpMethod::Delete
+            }
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct SecretRuleApiMatchValidation {
+    #[serde(rename = "type")]
+    pub r#type: String,
+    pub endpoint: Option<String>,
+    pub hosts: Option<Vec<String>>,
+    pub request_headers: Option<HashMap<String, String>>,
+    pub http_method: Option<SecretRuleApiMatchValidationHttpMethod>,
+    pub timeout_seconds: Option<u64>,
+    pub valid_http_status_code: Option<Vec<SecretRuleApiMatchValidationHttpCode>>,
+    pub invalid_http_status_code: Option<Vec<SecretRuleApiMatchValidationHttpCode>>,
+}
+
+impl From<SecretRuleApiMatchValidation> for SecretRuleMatchValidation {
+    fn from(val: SecretRuleApiMatchValidation) -> Self {
+        SecretRuleMatchValidation {
+            r#type: val.r#type,
+            endpoint: val.endpoint.clone(),
+            hosts: val.hosts.clone(),
+            request_headers: val.request_headers,
+            http_method: val.http_method.map(|v| v.into()),
+            timeout_seconds: val.timeout_seconds,
+            valid_http_status_code: val
+                .valid_http_status_code
+                .map(|v| v.iter().map(|w| w.clone().into()).collect()),
+            invalid_http_status_code: val
+                .invalid_http_status_code
+                .map(|v| v.iter().map(|w| w.clone().into()).collect()),
+        }
```

Comment:
no need to clone owned values:

```suggestion
        SecretRuleMatchValidation {
            r#type: val.r#type,
            endpoint: val.endpoint,
            hosts: val.hosts,
            request_headers: val.request_headers,
            http_method: val.http_method.map(|v| v.into()),
            timeout_seconds: val.timeout_seconds,
            valid_http_status_code: val
                .valid_http_status_code
                .map(|v| v.into_iter().map(|w| w.into()).collect()),
            invalid_http_status_code: val
                .invalid_http_status_code
                .map(|v| v.into_iter().map(|w| w.into()).collect()),
        }
```